File: app/src/main/java/com/chooloo/www/callmanager/ui/page/PageRecents.java
package com.chooloo.www.callmanager.ui.page;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

import com.chooloo.www.callmanager.ui.recents.RecentsFragment;

public class PageRecents extends PageFragment implements PageMvpView {

    private PageMvpPresenter<PageMvpView> mPresenter;

    private RecentsFragment mRecentsFragment;

    public static PageRecents newInstance() {
        Bundle args = new Bundle();
        PageRecents fragment = new PageRecents();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onSetup() {
        super.onSetup();

        mPresenter = new PagePresenter<>();
        mPresenter.attach(this);

        mRecentsFragment = RecentsFragment.newInstance();
        mRecentsFragment.setOnScrollListener(new RecyclerView.OnScrollListener() {
            @Override
            public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState) {
                super.onScrollStateChanged(recyclerView, newState);
                mPresenter.onScrollStateChanged(newState);
            }
        });

        _activity.getSupportFragmentManager().beginTransaction().replace(_binding.fragmentPageLayout.getId(), mRecentsFragment).commit();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mPresenter.detach();
    }

    @Override
    protected void onSearchTextChanged(String text) {
        mPresenter.onSearchTextChanged(text);
    }

    @Override
    protected void onDialNumberChanged(String number) {
        mPresenter.onDialNumberChanged(number);
    }

    @Override
    public void loadNumber(@Nullable String number) {
        mRecentsFragment.load(number, number == "" ? null : number);
    }

    @Override
    public void loadSearchText(@Nullable String text) {
        mRecentsFragment.load(null, text == "" ? null : text);
    }
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/page/PageRecents.kt
package com.chooloo.www.callmanager.ui.page

import com.chooloo.www.callmanager.ui.recents.RecentsFragment

class PageRecents : PageFragment(), PageMvpView {

    private lateinit var _presenter: PageMvpPresenter<PageMvpView>
    private lateinit var _recentsFragment: RecentsFragment

    companion object {
        fun newInstance(): PageRecents = PageRecents()
    }

    override fun onSetup() {
        super.onSetup()

        _presenter = PagePresenter()
        _presenter.attach(this)

        _recentsFragment = RecentsFragment.newInstance().apply {
            setOnScrollStateChangedListener { newState -> _presenter.onScrollStateChanged(newState) }
        }

        childFragmentManager.beginTransaction().replace(binding.fragmentPageLayout.id, _recentsFragment).commitNow()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _presenter.detach()
    }

    override fun onSearchTextChanged(text: String) {
        _presenter.onSearchTextChanged(text)
    }

    override fun onDialNumberChanged(number: String) {
        _presenter.onDialNumberChanged(number)
    }
}